ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -69,5 +87,84 @@ def authenticate(self, request, username=None, password=None, **kwargs):
                     "is_superuser": user.is_superuser,
                 },
             )
-
         return update_user_claims(user, self.database_instance, [])
+
+    def _can_authenticate_from_controller(self, username, password):
+        """
+        Check if a user exists in the AuthenticatorUser table with the local authenticator provider.
+        If the user is valid, update the gateway users credentials with the controller credentials.
+        """
+        try:
+            user = UserModel._default_manager.get_by_natural_key(username)
+        except UserModel.DoesNotExist:
+            logger.warning(f"User '{username}' does not exist in the database.")
+            return False
+
```

Comment:
```suggestion
if user.role.is_partially_migrated is False:
  return False
```

I expected to find a short-circuit before calling to the Controller API.

Reply:
This was removed due to @john-westcott-iv's comment here - https://github.com/ansible/django-ansible-base/pull/758#discussion_r2205103814

Reply:
Yea, we can double check with the upgrade patch but I'm pretty sure removing this flag was part of that. @AlanCoding might be able to confirm off the top of his head.

Reply:
Asked if `is_partially_migrated` is going away slack https://redhat-internal.slack.com/archives/C08NZHY16Q1/p1752586713478919

Reply:
Adding a note here for posterity - it was confirmed by @AlanCoding that `is_partially_migrated` is going away

Reply:
yep ping @TheRealHaoLiu